app/services/foreman_register/registration_token.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module ForemanRegister
+  class RegistrationToken
+    class << self
+      def encode(host, secret)
+        payload = prepare_payload(host, secret)
+        new JWT.encode(payload, secret)
+      end
+
+      private
+
+      def prepare_payload(host, secret)
+        exp = iat + (24 * 3600)
+        nbf = iat - 3600
+        jti_raw = [secret, iat].join(':')
+        jti = Digest::SHA256.hexdigest(jti_raw)
+        {
+          host_id: host.id,
+          iat: iat, # Issued At
+          jti: jti, # JWT ID
+          exp: exp, # Expiration Time
+          nbf: nbf # Not Before Time
+        }
+      end
+
+      def iat
```

Comment:
Could we use names? (also true for `prepare_payload`)
```suggestion
      def issued_at
```